cmd/state-svc/manager.go

```diff
@@ -130,11 +133,13 @@ func (s *serviceManager) CheckPid(pid int) (*int, error) {
 	}
 
 	// Try to verify that the matching pid is actually our process, because Windows aggressively reuses PIDs
-	exe, err := p.Exe()
-	if err != nil {
-		logging.Error("Could not detect executable for pid, error: %s", errs.JoinMessage(err))
-	} else if filepath.Clean(exe) != filepath.Clean(osutils.Executable()) {
-		return nil, nil
+	if runtime.GOOS == "windows" {
+		exe, err := p.Exe()
+		if err != nil {
+			logging.Error("Could not detect executable for pid, error: %s", errs.JoinMessage(err))
+		} else if filepath.Clean(exe) != filepath.Clean(osutils.Executable()) {
```

Comment:
Let's check the basename rather than the full path. Since we will also have people stranded on the old version where state-svc was at a different location.